test/integration/utils.go

```diff
@@ -9,20 +9,25 @@ import (
 	"testing"
 
 	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+	"github.com/stretchr/testify/require"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	kubernetesclient "k8s.io/client-go/kubernetes"
 
 	"github.com/kong/gateway-operator/pkg/consts"
 	"github.com/kong/gateway-operator/pkg/utils/kubernetes/resources"
+	"github.com/kong/gateway-operator/test/helpers"
 )
 
 // Expect404WithNoRouteFunc is used to check whether a given URL responds
 // with 404 and a standard Kong no route message.
 func Expect404WithNoRouteFunc(t *testing.T, ctx context.Context, url string) func() bool {
 	t.Helper()
 
+	httpClient, err := helpers.CreateHTTPClient(nil, "")
```

Comment:
Food for thought (we can leave this as is in this PR): this call site seems at the very least ambiguous :) My suggestion would be to reconsider the signature of `CreateHTTPClient` at some point.